app/controllers/contacts_controller.rb

```diff
@@ -29,6 +29,24 @@ def edit
     @contact = Contact.find(params[:id])
   end
 
+  def update
+    @contact = Contact.find(params[:id])
+    if @contact.update_attributes(contact_params)
+        redirect_to contacts_path(@contact), notice: "Contact Edited"
```

Comment:
this should either be a redirect to `contacts_path` (contact index) or `contact_path(@contact)` (contact show) - i think you want the 2nd one...